code/mob/living/critter/flockcritter_parent.dm

```diff
@@ -251,14 +251,15 @@
 	onUpdate()
 		..()
 		var/mob/living/critter/flock/F = owner
-		if (target == null || owner == null || !in_interact_range(owner, target) || isfeathertile(target) || !F?.can_afford(FLOCK_CONVERT_COST))
+		if (!target || !F || isdead(F) || !in_interact_range(F, target) || isfeathertile(target) || !F?.can_afford(FLOCK_CONVERT_COST))
 			interrupt(INTERRUPT_ALWAYS)
 			return
 
 	onStart()
 		..()
-		if(owner && target)
-			boutput(owner, "<span class='notice'>You begin spraying nanite strands onto the structure. You need to stay still for this.</span>")
+		var/mob/living/critter/flock/drone/F = owner
+		if(F && isalive(F) && target)
```

Comment:
surely the action should not be started if the flockdrone is dead?

Reply:
Looked at some other uses of onStart(), seems a lot of them call parent before checking if the action can happen, was guessing it might be just to show that the action happened but was interrupted, not sure otherwise.

Will make it actually interrupt

Reply:
Well in that case it'd probably make sense to put the common checks on a parent and call that, but it's relatively minor